File: Lib/Telerik/Examples/QuickStart/Sparkline/FirstLook/CallData.vb
Imports System
Imports System.ComponentModel

Namespace Telerik.Examples.WinControls.Sparkline.FirstLook
    Public Class CallData
        Implements INotifyPropertyChanged

        Private _duration As TimeSpan
        Private _holdTime As TimeSpan
        Private _issueResolved As Double
        Private timeInterval As DateTime
        Private _abandonments As Integer
        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

        Public Sub OnPropertyChanged(ByVal PropertyName As String)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(PropertyName))
        End Sub

        Public Property Abandonments As Integer
            Get
                Return Me._abandonments
            End Get
            Set(ByVal value As Integer)
                Me._abandonments = value
                Me.OnPropertyChanged("Abandonments")
            End Set
        End Property

        Public Property TimeInverval As DateTime
            Get
                Return Me.timeInterval
            End Get
            Set(ByVal value As DateTime)
                Me.timeInterval = value
                Me.OnPropertyChanged("TimeInverval")
            End Set
        End Property

        Public ReadOnly Property DurationSeconds As Integer
            Get
                Return CInt(Me.duration.TotalSeconds)
            End Get
        End Property

        Public ReadOnly Property DurationMinutes As Integer
            Get
                Return CInt(Me.duration.TotalMinutes)
            End Get
        End Property

        Public Property Duration As TimeSpan
            Get
                Return Me._duration
            End Get
            Set(ByVal value As TimeSpan)
                Me._duration = value
                Me.OnPropertyChanged("Duration")
            End Set
        End Property

        Public Property HoldTime As TimeSpan
            Get
                Return Me._holdTime
            End Get
            Set(ByVal value As TimeSpan)
                Me._holdTime = value
                Me.OnPropertyChanged("HoldTime")
            End Set
        End Property

        Public ReadOnly Property HoldTimeMinutes As Integer
            Get
                Return CInt(Me.holdTime.TotalMinutes)
            End Get
        End Property

        Public ReadOnly Property HoldTimeSeconds As Integer
            Get
                Return CInt(Me.holdTime.TotalSeconds)
            End Get
        End Property

        Public Property IssueResolved As Double
            Get
                Return Me._issueResolved
            End Get
            Set(ByVal value As Double)
                Me._issueResolved = value
                Me.OnPropertyChanged("IssueResolved")
            End Set
        End Property
    End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/Sparkline/FirstLook/CallData.cs
using System;
using System.ComponentModel;

namespace Telerik.Examples.WinControls.Sparkline.FirstLook
{
    public class CallData : INotifyPropertyChanged
    {
        private TimeSpan duration;
        private TimeSpan holdTime;
        private double issueResolved;
        private DateTime timeInterval;
        private int abandonments;

        public event PropertyChangedEventHandler PropertyChanged;

        public void OnPropertyChanged(string PropertyName)
        {
            if (PropertyChanged != null)
            {
                PropertyChanged(this, new PropertyChangedEventArgs(PropertyName));
            }
        }

        public int Abandonments
        {
            get
            {
                return this.abandonments;
            }
            set
            {
                this.abandonments = value;
                this.OnPropertyChanged("Abandonments");
            }
        }

        public DateTime TimeInverval
        {
            get
            {
                return this.timeInterval;
            }
            set
            {
                this.timeInterval = value;
                this.OnPropertyChanged("TimeInverval");
            }
        }

        public int DurationSeconds
        {
            get
            {
                return (int)this.Duration.TotalSeconds;
            }
        }

        public int DurationMinutes
        {
            get
            {
                return (int)this.Duration.TotalMinutes;
            }
        }

        public TimeSpan Duration
        {
            get
            {
                return this.duration;
            }
            set
            {
                this.duration = value;
                this.OnPropertyChanged("Duration");
            }
        }

        public TimeSpan HoldTime
        {
            get
            {
                return this.holdTime;
            }
            set
            {
                this.holdTime = value;
                this.OnPropertyChanged("HoldTime");
            }
        }

        public int HoldTimeMinutes
        {
            get
            {
                return (int)this.HoldTime.TotalMinutes;
            }
        }

        public int HoldTimeSeconds
        {
            get
            {
                return (int)this.HoldTime.TotalSeconds;
            }
        }

        public double IssueResolved
        {
            get
            {
                return this.issueResolved;
            }
            set
            {
                this.issueResolved = value;
                this.OnPropertyChanged("IssueResolved");
            }
        }
    }
}
